### PR TITLE
Show ihp details on the front end

### PR DESCRIPTION
**Pull request by hschallhorn**
Implements the front end of #15405

### Description
Adds a tool tip to queues to display information on ihp tasks associated with appeals

### Acceptance Criteria
- [x] Judge Assign queue shows the IHP timing details on hover
- [x] Atty Assigned queue shows the IHP timing details on hover
- [x] Attorney On Hold queue shows the IHP timing details on hover

### Testing Plan
1. Sign in as bvaaaabshire and click into a legacy appeal from your queue
1. Create an admin action for ihp
1. Complete the task as the colocated user
1. Reload the judge's queue
1. Hover over days waiting for the same case and confirm ihp was requested and received
- [x] Judge Assign queue shows the IHP timing details on hover for legacy
6. Repeat the same for an ama appeal
- [x] Judge Assign queue shows the IHP timing details on hover for ama
7. Sign in as bvacasper1 and select a legacy appeal from you assigned queue
1. Create an admin action for ihp
1. Confirm ihp request information shows up in your on hold tab
- [x] Attorney On Hold queue shows the IHP timing details on hover for legacy
10. Repeat the same for an ama appeal
- [x] Attorney On Hold queue shows the IHP timing details on hover for ama
11. Select a legacy appeal from your assigned queue and make some ihp tasks for it
```ruby
vacols_id = "2226048"
appeal = LegacyAppeal.find_by(vacols_id: vacols_id)
InformalHearingPresentationTask.create(
  assigned_to: Vso.first,
  created_at: 60.days.ago,
  appeal: appeal
).completed!
```
12. Confirm ihp request information shows up in your on assigned tab on reload
- [x] Atty Assigned queue shows the IHP timing details on hover for legacy
13. Repeat the same for ama
- [x] Atty Assigned queue shows the IHP timing details on hover for ama

### User Facing Changes
 - [x] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---
none|Screen Shot 2020-11-10 at 12 44 01 PM

### Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Storybook Story
* [x] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [x] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [x] Write a separate story (within the same file) for each discrete variation of the component

1. Start up your storybook server
```bash
make storybook
```
2. Navigate to http://localhost:6006/?path=/story/queue-components-ihpdayswaitingtooltip
3. Play with the new component! (still have not found a great way to change the dates nicely)

### CODE REVIEW

File: app/models/serializers/work_queue/legacy_task_serializer.rb [hunk 1]
```diff
@@ -91,4 +91,8 @@ class WorkQueue::LegacyTaskSerializer
   attribute :available_actions do |object, params|
     object.available_actions_unwrapper(params[:user], params[:role])
   end
+
+  attribute :latest_informal_hearing_presentation_task do |object|
+    { requested_at: object.assigned_at, recieved_at: object.started_at }
+  end
```
Comment (hschallhorn) [inline app/models/serializers/work_queue/legacy_task_serializer.rb:97]: Placeholder. Will remove

File: app/models/serializers/work_queue/task_column_serializer.rb [hunk 2]
```diff
@@ -248,6 +248,14 @@ def self.serialize_attribute?(params, columns)
     end
   end
 
+  attribute :latest_informal_hearing_presentation_task do |object, params|
+    columns = [Constants.QUEUE_CONFIG.COLUMNS.TASK_TYPE.name, Constants.QUEUE_CONFIG.COLUMNS.DAYS_WAITING.name]
+
+    if serialize_attribute?(params, columns)
+      { requested_at: object.assigned_at, received_at: object.started_at }
+    end
+  end
```
Comment (hschallhorn) [inline app/models/serializers/work_queue/task_column_serializer.rb:257]: Placeholder. Will remove

File: app/models/serializers/work_queue/legacy_task_serializer.rb [hunk 3]
```diff
@@ -95,6 +95,6 @@ class WorkQueue::LegacyTaskSerializer
   attribute :latest_informal_hearing_presentation_task do |object|
     task = object.appeal.latest_informal_hearing_presentation_task
 
-    task ? { requested_at: task.assigned_at, recieved_at: task.closed_at } : {}
+    task ? { requested_at: task.assigned_at, received_at: task.closed_at } : {}
```
Comment (hschallhorn) [inline app/models/serializers/work_queue/legacy_task_serializer.rb:98]: Was amazed I got through these PRs without this slip-up. Guess I was wrong...

File: client/app/queue/components/IhpDaysWaitingTooltip.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+const IhpDaysWaitingTooltip = (props) => {
+  const { requestedAt, receivedAt, children } = props;
+
+  if (!requestedAt) {
+    return children;
+  }
+
+  const today = moment();
+  const daysSinceIhpReceived = receivedAt ? `(${today.diff(moment(receivedAt), 'd')} days)` : '';
+  const daysWaiting = `${(receivedAt ? moment(receivedAt) : today).diff(moment(requestedAt), 'd')} days`;
```
Comment (hschallhorn) [inline client/app/queue/components/IhpDaysWaitingTooltip.jsx:30]: If we have received the IHP, use `receivedAt` as the "hold end date" to show how long we waited for the ihp. 
If we have not received the IHP, use today as the "hold end date" to show how long we have waited for the ihp so far.

File: client/app/queue/components/IhpDaysWaitingTooltip.stories.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+const Template = (args) => <IhpDaysWaitingTooltip {...args}>
+  <div style={{ textAlign: 'center' }}><span>3 days (hover over me)</span></div>
+</IhpDaysWaitingTooltip>;
+
+export const WaitingForIHP = Template.bind({});
+WaitingForIHP.args = { requestedAt: '2020-10-03T13:39:36.574-05:00', receivedAt: null };
+
+export const ReceivedIHP = Template.bind({});
+ReceivedIHP.args = { requestedAt: '2020-10-03T13:39:36.574-05:00', receivedAt: '2020-11-03T13:39:36.574-05:00' };
```
Comment (hschallhorn) [inline client/app/queue/components/IhpDaysWaitingTooltip.stories.js:24]: Woooo storybook! Check out PR body to find out how to run/play around

File: client/test/app/queue/components/IhpDaysWaitingTooltip.test.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        };
+
+        setup(propifyDates(dates));
+
+        expect(screen.getByTestId('ihp-received').textContent).toEqual('IHP Received:  ');
+        expect(screen.getByTestId('ihp-days-waiting').textContent).toEqual('On hold for IHP: 20 days');
+      });
+    });
+  });
+});
```
Comment (hschallhorn) [inline client/test/app/queue/components/IhpDaysWaitingTooltip.test.js:109]: First jest test! VERY open to thoughts!

File: client/test/app/queue/components/__snapshots__/IhpDaysWaitingTooltip.test.js.snap [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+            </strong>
+             
+            10 days
+          </li>
+        </ul>
+      </div>
+    </div>
+  </span>
+</div>
+`;
```
Comment (hschallhorn) [inline client/test/app/queue/components/__snapshots__/IhpDaysWaitingTooltip.test.js.snap:72]: Snapshot is created with [this test](https://github.com/department-of-veterans-affairs/caseflow/pull/15560/files#diff-e5e33c1be70b18bb66af30f52b2dc01f8c473df797155df8c48e3d60fd9d995bR60). This ensures we have a test that will fail if our ui changes at all 😍

File: client/test/app/queue/components/IhpDaysWaitingTooltip.test.js (hunk 6 above)
Comment (ajspotts) [reply client/test/app/queue/components/IhpDaysWaitingTooltip.test.js:109]: Very awesome! wish I had more experience with Jest to dissect this better, however looks pretty comprehensive to me and it's great we can throw in the a11y testing as well 🙌

**Review by ajspotts — APPROVED** [review]
Great changes and some sweet jest tests to accompany lgtm!

**Pull request merged**